src/game.cpp

```diff
@@ -9290,6 +9297,15 @@ void game::eat( int pos )
             item_loc.remove_item();
         }
     }
+    g->refresh_all();
+
+    // Set up activity to reopen this menu next turn
+    if( u.has_activity( activity_id( "ACT_EAT" ) ) ) {
+        u.activity.targets.clear();
```

Comment:
Need to make sure this doesn't have side effects. As we continue to increase the number of activities, clearing the whole list gets more and more risky.

Reply:
This doesn't clear the whole list of activity, u.activity.targets is just a vector of item_location the activity carries.
Currently, short of a bug, targets should never have more than one element here.